src/openrct2/world/Sprite.cpp

```diff
@@ -46,21 +46,13 @@ static std::array<std::vector<uint16_t>, SPATIAL_INDEX_SIZE> gSpriteSpatialIndex
 
 static void FreeEntity(EntityBase& entity);
 
-constexpr size_t GetSpatialIndexOffset(int32_t x, int32_t y)
+static constexpr size_t GetSpatialIndexOffset(const CoordsXY& loc)
 {
-    if (x == LOCATION_NULL)
-        return SPATIAL_INDEX_LOCATION_NULL;
-
-    const auto tileX = std::clamp<size_t>(x / COORDS_XY_STEP, 0, MAXIMUM_MAP_SIZE_TECHNICAL);
-    const auto tileY = std::clamp<size_t>(y / COORDS_XY_STEP, 0, MAXIMUM_MAP_SIZE_TECHNICAL);
-    const auto index = tileX * MAXIMUM_MAP_SIZE_TECHNICAL + tileY;
-
-    if (index >= std::size(gSpriteSpatialIndex))
-    {
+    if (loc.IsNull() || !map_is_location_valid(loc))
```

Comment:
Do we really need the additional `map_is_location_valid`? This function is going to be called super super often.

Reply:
It no longer clamps so it needs to check that.

Reply:
It turns out that this new form is also the fastest, see https://quick-bench.com/q/fHtaplRpsvlpiWH3dvulMwgR90A